src/main/java/org/ballerinax/datamapper/SampleDataAnalysisTask.java

```diff
@@ -333,6 +121,11 @@ public void pluginExecutionCompleted() {
         for (String moduleName : listOfModuleNames) {
             processSampleDataFiles(moduleName);
         }
+        if (!this.dataMapperLog.getDataMapperPluginDiagnostic().isEmpty()) {
+            for (Diagnostic diagnostic : this.dataMapperLog.getDataMapperPluginDiagnostic()) {
+                compilationAnalysisContext.reportDiagnostic(diagnostic);
+            }
```

Comment:
I'm assuming `getDataMapperPluginDiagnostic()` returns a list? If so, there's no need for the if check here.

Reply:
Removed the if check